docs/tutorials/add-payment-adapter.md

```diff
@@ -0,0 +1,204 @@
+     */
+    public function getPaymentMethod(string $customerId, string $paymentMethodId): array
+    {
+        
+    }
+
+    /**
+     * Update card
+     */
+    public function updatePaymentMethodBillingDetails(string $paymentMethodId, string $name = null, string $email = null, string $phone = null, array $address = null): array
+    {
+        
+    }
+
+    public function updatePaymentMethod(string $paymentMethodId, string $type, array $details): array
+    {
+        
+    }
+
+    /**
+     * Delete a credit card record
+     */
+    public function deletePaymentMethod(string $paymentMethodId): bool
+    {
+        
+    }
+
+    /**
+     * Add new customer in the gateway database
+     * returns the newly created customer
+     *
+     * @throws Exception
+     */
+    public function createCustomer(string $name, string $email, array $address = [], string $paymentMethod = null): array
+    {
+        
+    }
+
+    /**
+     * List customers
+     */
+    public function listCustomers(): array
+    {
+        
+    }
+
+    /**
+     * Get customer details by ID
+     */
+    public function getCustomer(string $customerId): array
+    {
+        
+    }
+
+    /**
+     * Update customer details
+     */
+    public function updateCustomer(string $customerId, string $name, string $email, array $address = null, string $paymentMethod = null): array
+    {
+        
+    }
+
+    /**
+     * Delete customer by ID
+     */
+    public function deleteCustomer(string $customerId): bool
+    {
+        
+    }
+
+    /**
+     * Save Payment Method from Front-end
+     */
+    public function createFuturePayment(string $customerId, array $paymentMethodTypes): array
+    {
+        
+    }
+}
+
+```
+
+> If you copy this template, make sure to replace all placeholders wrapped like `[THIS]` and to implement everything marked as `TODO:`.
+
+When implementing new adapter, please make sure to follow these rules:
+
+- `getName()` needs to use same name as file name with first letter lowercased. For example, in `Stripe.php`, we use `stripe`
+
+
+Please mention in your documentation what resources or API docs you used to implement the provider's API.
+
+## 2. Test your adapter
+
+After you finished adding your new adapter, you should write a proper test for it. To do that, you create `tests/Pay/Adapter/[PROVIDER_NAME]Tests.php` and write tests for the provider. Look at the test written for the existing adapters for examples.
+
+To run the test you can simply run
+
+```bash
+composer test
+```
+
+## 4. Raise a pull request
+
+First of all, commit the changes with the message `Added XXX Adapter` (where `XXX` is adapter name) and push it. This will publish a new branch to your forked version of utopia/pay. If you visit it at `github.com/YOUR_USERNAME/pay`, you will see a new alert saying you are ready to submit a pull request. Follow the steps GitHub provides, and at the end, you will have your pull request submitted.
```

Comment:
```suggestion
First of all, commit the changes with the message `Add XXX Adapter` (where `XXX` is adapter name) and push it. This will publish a new branch to your forked version of utopia-php/pay. If you visit `github.com/YOUR_USERNAME/pay`, you will see a new alert saying you are ready to submit a pull request. Follow the steps GitHub provides, and at the end, you will have your pull request submitted.
```